core/spring-boot/src/main/java/org/springframework/boot/logging/log4j2/SpringBootTriggeringPolicy.java

```diff
@@ -0,0 +1,248 @@
+import org.apache.logging.log4j.core.config.Configuration;
+import org.apache.logging.log4j.core.config.Node;
+import org.apache.logging.log4j.core.config.plugins.Plugin;
+import org.apache.logging.log4j.core.config.plugins.PluginAttribute;
+import org.apache.logging.log4j.core.config.plugins.PluginBuilderFactory;
+import org.apache.logging.log4j.core.config.plugins.PluginConfiguration;
+import org.apache.logging.log4j.core.config.plugins.PluginFactory;
+import org.apache.logging.log4j.core.util.Builder;
+import org.jspecify.annotations.Nullable;
+
+import org.springframework.util.Assert;
+
+/**
+ * {@link TriggeringPolicy} that selects one of several standard Log4j2
+ * {@link TriggeringPolicy TriggeringPolicies} based on configuration attributes. The
+ * supported strategies are {@code size}, {@code time}, {@code size-and-time}, and
+ * {@code cron}.
+ *
+ * @author HoJoo Moon
+ * @since 4.0.0
+ */
+@Plugin(name = "SpringBootTriggeringPolicy", category = Node.CATEGORY, elementType = "TriggeringPolicy",
+		deferChildren = true, printObject = true)
+public final class SpringBootTriggeringPolicy extends AbstractTriggeringPolicy {
+
+	private final TriggeringPolicy delegate;
+
+	private SpringBootTriggeringPolicy(TriggeringPolicy delegate) {
+		this.delegate = delegate;
+	}
+
+	TriggeringPolicy getDelegate() {
+		return this.delegate;
+	}
+
+	@Override
+	public void initialize(RollingFileManager manager) {
+		this.delegate.initialize(manager);
+	}
+
+	@Override
+	public boolean isTriggeringEvent(LogEvent event) {
+		return this.delegate.isTriggeringEvent(event);
+	}
+
+	@Override
+	public void start() {
+		super.start();
+		if (this.delegate instanceof LifeCycle lifecycle) {
+			lifecycle.start();
+		}
+	}
+
+	@Override
+	public boolean stop(long timeout, TimeUnit timeUnit) {
+		setStopping();
+		boolean result = true;
+		if (this.delegate instanceof LifeCycle2 lifecycle2) {
+			result = lifecycle2.stop(timeout, timeUnit);
+		}
+		else if (this.delegate instanceof LifeCycle lifecycle) {
+			lifecycle.stop();
+		}
+		setStopped();
+		return result;
+	}
+
+	@Override
+	public boolean isStarted() {
+		if (this.delegate instanceof LifeCycle lifecycle) {
+			return lifecycle.isStarted();
+		}
+		return super.isStarted();
+	}
+
+	@Override
+	public String toString() {
+		return "SpringBootTriggeringPolicy{" + this.delegate + "}";
+	}
+
+	@PluginBuilderFactory
+	public static SpringBootTriggeringPolicyBuilder newBuilder() {
+		return new SpringBootTriggeringPolicyBuilder();
+	}
+
+	@PluginFactory
+	public static SpringBootTriggeringPolicy createPolicy(@PluginAttribute("strategy") @Nullable String strategy,
+			@PluginAttribute("maxFileSize") @Nullable String maxFileSize,
+			@PluginAttribute("timeInterval") @Nullable Integer timeInterval,
+			@PluginAttribute("timeModulate") @Nullable Boolean timeModulate,
+			@PluginAttribute("cronExpression") @Nullable String cronExpression,
+			@PluginConfiguration Configuration configuration) {
```

Comment:
There is no need for both a factory method and a builder. Since the builder is easier to maintain than the factory method, you should leave just the builder.